back/src/chat/chat.service.ts

```diff
@@ -1,51 +1,40 @@
 import { BadRequestException, Injectable } from '@nestjs/common';
-import { Server } from 'socket.io';
 import Socket from '@type/socket';
+import type { Server } from '@type/server';
 
 @Injectable()
 export class ChatService {
 	server: Server;
-	clientMap: Map<number, string[]> = new Map();
 
-	addSocket(socket: Socket) {
-		if (!this.clientMap.has(socket.user.id)) this.clientMap.set(socket.user.id, [socket.id]);
-		else this.clientMap.get(socket.user.id).push(socket.id);
+	async fetchSockets(userId: number) {
+		const sockets = await this.server.in(`user:${userId}`).fetchSockets();
+		return { sockets, socketIds: sockets.map((socket) => socket.id) };
 	}
 
-	removeSocket(socket: Socket) {
-		if (this.clientMap.has(socket.user.id)) {
-			const sockets: string[] = this.clientMap.get(socket.user.id).filter((s: string) => s !== socket.id);
-			if (sockets.length === 0) this.clientMap.delete(socket.user.id);
-			else this.clientMap.set(socket.user.id, sockets);
-		}
-	}
-
-	onConnection(socket: Socket) {
-		this.addSocket(socket);
-		console.log(`${socket.user.username}: `, this.clientMap.get(socket.user.id));
-	}
-
-	onDisconnection(socket: Socket) {
-		this.removeSocket(socket);
-		console.log(`${socket.user.username}: `, this.clientMap.get(socket.user.id));
-	}
-
-	emitToUser(userId: number, event: string, data: any): void {
-		if (this.clientMap.has(userId)) this.server.to(this.clientMap.get(userId)).emit(event, data);
+	async onConnection(socket: Socket) {
+		socket.join(`user:${socket.user.id}`);
+		const { socketIds } = await this.fetchSockets(socket.user.id);
+		console.log(`${socket.user.username} (onConnection): `, socketIds);
 	}
 
 	emit(event: string, data: any, rooms?: string | string[]) {
 		if (rooms) this.server.to(rooms).emit(event, data);
 		else this.server.emit(event, data);
 	}
 
-	joinRoom(socket_id: string, room: string) {
-		const socket: Socket = (this.server.sockets as any as Map<string, Socket>).get(socket_id);
+	emitToUser(userId: number, event: string, data: any): void {
```

Comment:
I would have suggested
``emitToUser(event: string, data: any, userId: number): void {``
instead of
``emitToUser(userId: number, event: string, data: any): void {``

So the arguments are in the same order as they are for emit and only the function name changes

Reply:
Done!